batch/batch/server/server.py

```diff
@@ -216,56 +214,36 @@ def _has_next_task(self):
-                HAIL_POD_NAMESPACE,
-                pod_template,
-                _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
-            self._pod_name = pod.metadata.name
-
-            await db.jobs.update_record(self.id,
-                                        pod_name=self._pod_name)
-
-            log.info('created pod name: {} for job {}, task {}'.format(self._pod_name,
-                                                                       self.id,
-                                                                       self._current_task.name))
-        except kube.client.rest.ApiException as err:
+        pod, err = await app['k8s'].create_pod(body=pod_template)
+        if err is not None:
+            traceback.print_tb(err.__traceback__)
             log.info(f'pod creation failed for job {self.id} with the following error: {err}')
+            return
+        self._pod_name = pod.metadata.name
+        await db.jobs.update_record(self.id,
+                                    pod_name=self._pod_name)
+        log.info('created pod name: {} for job {}, task {}'.format(self._pod_name,
+                                                                   self.id,
+                                                                   self._current_task.name))
 
     async def _delete_pvc(self):
         if self._pvc_name is None:
             return
 
         log.info(f'deleting persistent volume claim {self._pvc_name}')
-        try:
-            v1.delete_namespaced_persistent_volume_claim(
-                self._pvc_name,
-                HAIL_POD_NAMESPACE,
-                _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
-        except kube.client.rest.ApiException as err:
-            if err.status == 404:
-                log.info(f'persistent volume claim {self._pvc_name} is already deleted')
-                return
-            raise
-        finally:
-            await db.jobs.update_record(self.id, pvc_name=None)
-            self._pvc_name = None
+        err = await app['k8s'].delete_pvc(self._pvc_name)
+        if err is not None:
+            raise ValueError('could not delete {self._pvc_name}') from err
+        await db.jobs.update_record(self.id, pvc_name=None)
+        self._pvc_name = None
 
     async def _delete_k8s_resources(self):
         await self._delete_pvc()
         if self._pod_name is not None:
-            try:
-                v1.delete_namespaced_pod(
-                    self._pod_name,
-                    HAIL_POD_NAMESPACE,
-                    _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
-            except kube.client.rest.ApiException as err:
-                if err.status == 404:
-                    pass
-                raise
-            finally:
-                await db.jobs.update_record(self.id, pod_name=None)
-                self._pod_name = None
+            await db.jobs.update_record(self.id, pod_name=None)
+            await app['k8s'].delete_pod(name=self._pod_name,
```

Comment:
Why do we not check or log the error code here?

Reply:
I went through and made sure we at least logged errors when they occurred. There's a few cases where we should probably think more carefully about what an error means.